### DISCUSSION TITLE
Do we have a roadmap or an idea what should DART be and what it should not be?

### DISCUSSION THREAD

**Discussion by olzhas** [dartsim/dart#1780] [General]
title

**Comment by jslee02** [comment]
Sorry this sat unanswered for so long. This may no longer be needed by the original poster, but for anyone else with the same question:

Yes, DART now has a clearer roadmap and project direction documented in the repository. The short version is that DART is a research-focused physics engine for robotics, animation, and machine learning, with priorities around accurate articulated-body dynamics, trustworthy contact/constraint solving, broad model loading, and an easy public API on a maintainable C++20 core.

The current direction is split roughly like this:

- DART 7/main is the active clean-break line, focused on an easier-start research API, Python-first workflows, algorithm extension points, and scalable compute work.
- DART 6.x release branches are compatibility/maintenance lines for the established API and downstream users; normal new DART 7 features are not backported there by default.
- Roadmap state lives in `docs/plans/dashboard.md`, with strategic framing in `docs/ai/north-star.md` and `docs/plans/north-star-roadmap.md`.

What DART should not be: a black-box-only simulator, a place where every experimental backend or solver detail becomes public API immediately, or a project where DART 6 compatibility blocks the DART 7 API cleanup. New features should come with clear user value, tests/benchmarks where appropriate, and a maintainable public surface.

Useful links:
- https://github.com/dartsim/dart/blob/main/docs/ai/north-star.md
- https://github.com/dartsim/dart/blob/main/docs/plans/dashboard.md
- https://github.com/dartsim/dart/blob/main/docs/onboarding/release-roadmap.md
